pkg/azuremachinepool/validate_azuremachinepool_update_test.go

```diff
@@ -41,92 +41,110 @@ func TestAzureMachinePoolUpdateValidate(t *testing.T) {
 	testCases := []testCase{
 		{
 			name:         "case 0: AcceleratedNetworking is enabled in CR and we don't change it or the instance type",
-			oldNodePool:  azureMPRawObject(supportedInstanceType[0], &tr, string(compute.StorageAccountTypesStandardLRS)),
-			newNodePool:  azureMPRawObject(supportedInstanceType[0], &tr, string(compute.StorageAccountTypesStandardLRS)),
+			oldNodePool:  azureMPRawObject(supportedInstanceType[0], &tr, string(compute.StorageAccountTypesStandardLRS), nil),
```

Comment:
What do you think about passing `desiredDatadisks` instead of `nil` and removing the `nil` check from the function? I feel that always passing the disks is more explicit and we don't have to know that `nil` means `desiredDatadisks`.